### PR TITLE
Milestone2: Feature-Folder Structure

### PR DESCRIPTION
**Pull request by vanheaven-ui**
## This PR seeks a review of the folder structure layout and basic implementation of the redux store for the project React components.

Below is what was executed to accomplish the code herein:
- Created folders in the src directory and these are; actions, components, containers, and reducers.
- Implemented the necessary components in their respective folders.
- Implemented the redux actions for books.
- Implemented the redux reducer for books.
- Initialized the redux store.
- Connected the high-level component to redux store through the Provider component.

**NOTE:** 
- The useSelector redux hook was used to map state to props that are accessed by the React components
- The implementation of this PR was deployed to Heroku.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import { Provider } from 'react-redux';
+import App from './components/App';
+import store from './store';
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Provider store={store}>
+      <App />
+    </Provider>
```
Comment (trekab) [inline src/index.js]: After wrapping the `App` in `Provider` it is cleaner to assign the whole thing into a variable e.g. 
```javascript
const wrappedApp = (
  <Provider store={store}>
    <App />
  </Provider>
);
```
What do you think? :thinking:

**Review by trekab — CHANGES_REQUESTED** [review]
#### Changes Required! :recycle: 
Hi @vanheaven-ui! So far so good! :+1: There are just a few details I'd like you to look at before completing this milestone. These include:
- [ ] Example books should be added to the initial state object via the `src/index.js` file. ([screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/InitialBooks.png))
- [ ] You did well connecting to the redux store using `react-redux's` hook i.e. `useSelector` :+1: However, on this milestone making use of `mapStateToProps` to fetch books from the store in the `BooksList` component is a requirement.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

File: src/index.js (hunk 1 above)
Comment (vanheaven-ui) [reply src/index.js]: Haahaha!.. It is quite readable but at the expense of memory.

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi @vanheaven-ui :hand: 
Awesome job on this milestone.
Good luck with the next one

Happy coding :computer: 
**Sérgio Torres**

**Pull request closed without merging**